src/highlight/highlight.ts

```diff
@@ -1,59 +1,35 @@
-import { Component, Input } from '@angular/core';
-import { bem, getPropertyByPath } from '../utils';
+import { ChangeDetectionStrategy, Component, Input } from '@angular/core';
+
+import { Hit } from 'instantsearch.js';
+
+import { highlight } from 'instantsearch.js/es/helpers';
+import { getPropertyByPath } from 'instantsearch.js/es/lib/utils';
 
 @Component({
   selector: 'ais-highlight',
-  template: `<span [class]="cx()" [innerHtml]="content"></span>`,
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  template: `<span class="ais-Highlight" [innerHtml]="content"></span>`,
 })
 export class NgAisHighlight {
   @Input() attribute: string;
-  @Input() hit: { _highlightResult?: {}; label?: string; highlighted?: string };
-  @Input() tagName: string = 'em';
-
-  cx = bem('Highlight');
+  @Input() hit: Partial<Hit>;
+  @Input() tagName: string = 'mark';
 
   get content() {
-    if (this.attribute === 'highlighted') {
-      return this.hit.highlighted
-        ? this.replaceWithTagName(this.hit.highlighted)
-        : this.hit.label;
-    }
-
-    if (this.hit.hasOwnProperty('_highlightResult')) {
-      const attributeHighlighted = getPropertyByPath(
-        this.hit._highlightResult,
-        this.attribute
-      );
-
-      // check that the attributeHighlighted is a string
-      if (
-        attributeHighlighted !== undefined &&
-        typeof attributeHighlighted.value === 'string'
-      ) {
-        return this.replaceWithTagName(attributeHighlighted.value);
-      }
-    }
-
+    const highlightAttributeResult = getPropertyByPath(
+      this.hit._highlightResult,
+      this.attribute
+    );
     const fallback = getPropertyByPath(this.hit, this.attribute);
-    if (!fallback) {
-      console.warn(
-        `Could not find attribute [${
-          this.attribute
-        }] into hit object, will display an empty string.`
-      );
 
-      return '';
+    if (!highlightAttributeResult && fallback) {
+      return fallback;
     }
```

Comment:
Isn't this more correct? (And it only computes the fallback when needed.)

```suggestion
    if (!highlightAttributeResult) {
      return getPropertyByPath(this.hit, this.attribute);
    }
```

Reply:
This is a bit trickier. Previous behaviour in Angular was this:
- return highlightResult, or
- return fallback value (attribute value), or
- show warning and return empty string

IS.js currently does this:
- return highlightResult, or
- show warning and return empty string

In this new implementation want the highlight to come from IS.js but also the warning, so we just need to intervene when there is no highlightResult **but** there is a fallback (so that we still have the warning from IS.js if both are undefined).

I agree that there's a cost by computing `fallback` unconditionnally, but I could not find a better way to do this without also replicating the warning in this implementation.